sktime/forecasting/rbf_forecaster.py

```diff
@@ -0,0 +1,575 @@
+
+        Parameters
+        ----------
+        y : pd.Series
+            The time series data to be fitted.
+        X : optional
+            Additional exogenous data (not used).
+        fh : optional
+            Forecasting horizon (not used).
+        """
+        import torch
+        from torch.utils.data import DataLoader, TensorDataset
+
+        self._y = y.copy()
+        if isinstance(y, pd.Series):
+            y_scaled = self.scaler.fit_transform(y.values.reshape(-1, 1))
+        else:
+            y_scaled = self.scaler.fit_transform(y.values)
+
+        X_train, y_train = self._create_windows(y_scaled)
+        X_tensor = torch.FloatTensor(X_train).to(self.device)
+        y_tensor = torch.FloatTensor(y_train).to(self.device)
+
+        dataset = TensorDataset(X_tensor, y_tensor)
+        dataloader = DataLoader(dataset, batch_size=self.batch_size, shuffle=True)
+
+        input_size = self.window_length
+        self.build_network(input_size)
+
+        criterion = self._instantiate_criterion()
+        optimizer = None
+
+        self.model.train()
+        for epoch in range(self.epochs):
+            total_loss = 0
+            for batch_X, batch_y in dataloader:
+                if optimizer is None:
+                    _ = self.model(batch_X)
+                    optimizer = self._instantiate_optimizer()
+
+                optimizer.zero_grad()
+                outputs = self.model(batch_X)
+                loss = criterion(outputs, batch_y)
+                loss.backward()
+                optimizer.step()
+                total_loss += loss.item()
+
+        return self
+
+    def _predict(self, fh, X=None):
+        """Generate predictions for the specified forecasting horizon.
+
+        Parameters
+        ----------
+        fh : ForecastingHorizon
+            The forecasting horizon for the predictions.
+        X : optional
+            Additional exogenous data (not used).
+        """
+        import torch
+
+        fh_abs = fh.to_absolute(self._y.index[-1]).to_numpy()
+        n_steps = len(fh_abs)
+        predictions = np.zeros(n_steps)
+
+        last_window = self._y[-self.window_length :].copy()
+        current_window = self.scaler.transform(last_window.values.reshape(-1, 1))
+
+        self.model.eval()
+        with torch.no_grad():
+            for i in range(n_steps):
+                X_predict = torch.FloatTensor(current_window.reshape(1, -1)).to(
+                    self.device
+                )
+                pred_scaled = self.model(X_predict).cpu().numpy()
+                pred = self.scaler.inverse_transform(pred_scaled.reshape(1, -1))
+                predictions[i] = pred.ravel()[0]
+
+                current_window = np.roll(current_window, -1, axis=0)
+                current_window[-1] = pred_scaled.ravel()[0]
+
+        return pd.Series(predictions, index=fh_abs, name=self._y.name)
+
+    def plot_predictions(self, y_train, y_test, y_pred):
```

Comment:
Why did you add this plot method? 

I think the `sktime` plotting methods that are not bound to specific estimators are sufficient.

Reply:
thought it might be useful that the forecaster have its own plotting method😅 I will remove it